### DISCUSSION TITLE
Getting relation multiple field from `pb.js`

### DISCUSSION THREAD

**Discussion by benstigsen** [pocketbase/pocketbase#5763] [Q&A] [answered]
Currently I have:

Post:
- title
- tags (relation multiple)

Tag:
- name

Does it require two separate queries to get the tags? Because I'm not able to "expand" the tags it seems.
```js
$app.dao().db()
  .newQuery("SELECT title, tags FROM posts")
  .all(posts);
```

`posts[0].tags` here is something like `["x8321hf7", "d9sxzn4"]`

**Comment by ganigeorgiev** [comment, marked as answer]
There is no automatic relations expansion when using PocketBase as framework (_especially with raw queries_).

There are not enough details to understand the full picture of your use case but Yes - you can perform separate query(ies) to retrieve the related tags.

Depending on what actually you are trying to do, an easier alternative could be to use the [Record model helpers](https://pocketbase.io/docs/js-records/) (_and more specifically [Programmatically expanding relations](https://pocketbase.io/docs/js-records/#programmatically-expanding-relations)_). For example:

```js
// note: in the upcoming v0.23 this is renamed to findAllRecords
const posts = $app.dao().findRecordsByExpr("posts", /* no extra expressions */)

$app.dao().expandRecords(posts, ["tags"])

// you can either operate with the Record model helpers or convert the result into a plain array
// (the related tags will be under the "expand.tags" key, the same way as in the Web APIs)
const plainPosts = JSON.parse(JSON.stringify(posts))
```

**Reply by ganigeorgiev** [reply]
_Side-note: The last `JSON.parse(JSON.stringify)` could be written slightly more efficiently as:_
```js
const plainPosts = posts.map((post) => post.publicExport());
```
